src/main.rs

```diff
@@ -37,6 +37,44 @@ fn main() -> anyhow::Result<()> {
             .info(CompileTimeInformation::default())
             .print::<Markdown>();
         std::process::exit(0);
+    } else if args.list {
```

Comment:
> Is it worth adding a check that makes sure unexpected CLI use like --list --explain foo gets noticed and reported? Right now, it seems like --list dominates almost everything else and causes it to be silently ignored.

I can go either way.  I had originally written it treating the flags as if they were overwriting each other (which we can explicitly have clap do on our behalf so intent is clear and to make sure  the last one wins)

Reply:
Since some things are flags and some are arguments (`check-release`), I'm worried about weird cases like `cargo semver-checks check-release --list` which might non-obviously exit 0 through the completely unintended outcome of not actually checking the release.

I'm inclined to say we should probably return errors for unexpected cases rather than doing last-one-wins.

Reply:
> `cargo semver-checks check-release --list`

I intentionally supported that with `global = true` as these seemed like universally applicable flags.

For mixing them together though, I've gone ahead and made them exclusive